javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/model/resolution/SymbolReference.java

```diff
@@ -33,54 +34,81 @@
  */
 public class SymbolReference<S extends ResolvedDeclaration> {
 
-    private Optional<? extends S> correspondingDeclaration;
+    private final S correspondingDeclaration;
 
-    private SymbolReference(Optional<? extends S> correspondingDeclaration) {
+    private SymbolReference(@Nullable S correspondingDeclaration) {
         this.correspondingDeclaration = correspondingDeclaration;
     }
 
     /**
      * Create a solve reference to the given symbol.
      */
     public static <S extends ResolvedDeclaration, S2 extends S> SymbolReference<S> solved(S2 symbolDeclaration) {
-        return new SymbolReference<S>(Optional.of(symbolDeclaration));
+        return new SymbolReference<>(symbolDeclaration);
+    }
+
+    /**
+     * Create a reference for an unsolved symbol.
+     *
+     * @return The created unsolved symbol reference.
+     *
+     * @param <S> The symbol reference type.
+     */
+    public static <S extends ResolvedDeclaration> SymbolReference<S> unsolved() {
+        return new SymbolReference<>(null);
     }
 
     /**
      * Create an unsolved reference specifying the type of the value expected.
      */
     public static <S extends ResolvedDeclaration, S2 extends S> SymbolReference<S> unsolved(Class<S2> clazz) {
-        return new SymbolReference<>(Optional.empty());
+        return new SymbolReference<>(null);
```

Comment:
Can this method declaration be simplified?

Reply:
It's a good question. Simplifying is not easy, since it impacts the api.
I think the best approach would be leave it as it is, and consider it as deprecated :)